net/dns/manager_windows.go

```diff
@@ -308,23 +300,25 @@ func (m windowsManager) Close() error {
 // Windows DHCP client from sending dynamic DNS updates for our interface to
 // AD domain controllers.
 func (m windowsManager) disableDynamicUpdates() error {
-	setRegValue := func(regBase string) error {
-		key, err := m.openKey(m.ifPath(regBase))
-		if err != nil {
+	for _, p := range []winutil.InterfaceRegKeyPrefix{winutil.IPv4TCPIPInterfacePrefix, winutil.IPv4TCPIPInterfacePrefix} {
+		if err := m.setSingleDWORD(p, "EnableDNSUpdate", 0); err != nil {
 			return err
 		}
-		defer key.Close()
-
-		return key.SetDWordValue("DisableDynamicUpdate", 1)
 	}
 
-	for _, regBase := range []string{ipv4RegBase, ipv6RegBase} {
-		if err := setRegValue(regBase); err != nil {
-			return err
-		}
+	return nil
+}
+
+func (m windowsManager) setSingleDWORD(p winutil.InterfaceRegKeyPrefix, value string, data uint32) error {
```

Comment:
on what interface? it takes an interface prefix, but then ... is the "value string" the interface name? docs?